.github/workflows/delegator_generic_slsa3.yml

```diff
@@ -0,0 +1,257 @@
+# Copyright 2022 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA builder delegator
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  SLSA_OUTPUTS_DIR: __SLSA_OUTPUTS_DIR__
+  SLSA_ARTIFACTS_FILE: artifacts-layout.json
+  SLSA_PREDICATE_FILE: predicate.json
+
+on:
+  workflow_call:
+    # Secrets provided as arguments are passed as inputs to the callback Action.
+    secrets:
+      secret1:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+      secret2:
+        description: >
+          Optional secret.
+
+          This argument is passed, unchanged, to the builder's Action's `build-artifacts`.
+        required: false
+
+    # Inputs provided as arguments are passed as inputs to the callback Action,
+    # formatted as a map.
+    inputs:
+      slsa-token:
+        description: "The SLSA token identifying the request"
+        required: true
+        type: string
+
+    # Outputs are provided to the caller TRW.
+    outputs:
+      # This contains all the ouputs of the callback Action, formatted as a map.
+      build-artifacts-outputs:
+        description: "The outputs from the build-artitacts Action, unchanged."
+        value: ${{ jobs.build-artifacts-ubuntu.outputs.outputs }}
+      
+      # This is an output from the framework.
+      attestations-download-name:
+        description: "Name of the artifact to download all the attestations."
```

Comment:
Either in the comment or you can defer to some docs that this folder will contain `.intoto` attestations when run in pre-submit without token access, or signed attestations `intoto.sigstore`.

Reply:
I'll add this in the description. Relevant for TRW writers. Thanks